### PR TITLE
Release process improvements

### PR DESCRIPTION
**Pull request by ddbeck**
This cleans up some annoyances around releasing the package weekly, by

- breaking up the monolithic release notes script into two smaller utilities: one that generates release stats and another that gives you the link to PRs that need release notes
- rewriting the publishing docs to reflect my current practice of using a draft PR
- improving the output from the `statistics` script, to make it easier to copy and paste into applicable issues

Notably, the release notes stats counts the number of features using the experimental `walk()` generator from `utils`, instead of duplicating the count.

@Elchi3 I've tagged you for review since I thought you might like to know about these changes, but your review is optional. Since I'm the only one doing this routinely, I figured I'd be the only one negatively impacted if there was a problem.

### CODE REVIEW

File: docs/publishing.md [hunk 1]
```diff
@@ -11,45 +11,66 @@ Any project owner (or release designee) can complete the following steps to publ
...
 
-1. Figure out the new version number by looking at [past releases](https://github.com/mdn/browser-compat-data/releases). If the release is a non-breaking and data-only update, we're using patch versions. Lets assume that's the case and the next version should be `1.0.3`.
+Start preparing a release when:
 
-2. On your updated and clean `main` branch, run `npm version patch -m "Patch release containing data or non-breaking updates only"`. Locally, this updates `package.json`, creates a new commit, and creates a new release tag (see also the docs for [npm version](https://docs.npmjs.com/cli/version)).
+- the first pull request labeled [needs-release-note 📰](https://github.com/mdn/browser-compat-data/pulls?q=is%3Apr+label%3A%22needs-release-note+%3Anewspaper%3A%22+) has been merged after the most-recent release
+- you're ready to issue a new release
 
-3. Push the commit to `main`: `git push origin main`.
+whichever comes first first.
```
Comment (Elchi3) [inline docs/publishing.md:24]:
```suggestion
whichever comes first.
```

File: docs/publishing.md [hunk 2]
```diff
@@ -11,45 +11,66 @@ Any project owner (or release designee) can complete the following steps to publ
...
 
-4. Check if the commit passes our [GitHub Actions workflows](https://github.com/mdn/browser-compat-data/actions).
+1. Start a new branch for the upcoming release. For example, run `git switch -c release-YYYY-MM-DD`, where `YYYY-MM-DD` is the target release date.
 
-5. If the commit passed, push the git tag as well: `git push origin v1.0.3`.
```
Comment (Elchi3) [inline docs/publishing.md:28]: Where in the new process is the git tag created?

**Review by Elchi3 — APPROVED** [review]
Nice! I haven't tested the scripts. I guess either it will work next release or it won't and then you will have to fix :) Just one question on how tags are created these days as it wasn't obvious to me from glancing over the new text.

File: docs/publishing.md (hunk 2 above)
Comment (ddbeck) [reply docs/publishing.md:28]: Oh I suppose this is a bit surprising. It happens in the GitHub UI, when you publish the release (step 2, in _Publish to npm_). I add a little bit of text to make this bit more explicit.

Comment (Elchi3) [reply docs/publishing.md:28]: Interesting! :) I've never released BCD with GitHub Actions, so this was a nice "today I learned". Thanks for the update!

**Comment by Elchi3** [comment]
LGTM, please merge when you feel like it.

**Comment by ddbeck** [comment]
@Elchi3 thank you!

**Pull request merged**